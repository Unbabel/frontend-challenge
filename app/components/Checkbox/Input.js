import styled from 'styled-components';

const Input = styled.input`
  position: absolute;
  left: -99999px;
`;

export default Input;
