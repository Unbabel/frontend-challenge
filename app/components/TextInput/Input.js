import styled from 'styled-components';

const Input = styled.input`
  width: 100%;
  border: none;
  border-bottom: 2px solid transparent;
  outline: none;
  box-sizing: border-box;
  padding: 4px;
  color: ${props => props.theme.transcriptionTitleColor};
  font-weight: 600;
  font-family: 'Montserrat', sans-serif;
  letter-spacing: 0;

  &:focus {
    border-bottom: ${props => `2px solid ${props.theme.blue}`};
  }
`;

export default Input;
