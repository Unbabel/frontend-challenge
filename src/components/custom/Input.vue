<template>
    <div contenteditable @input="updateValue">{{ modelValue }}</div>
</template>

<script lang="ts">
import { defineComponent } from 'vue'

export default defineComponent({
    props: ['modelValue'],
    emits: ['update:modelValue'],
    methods: {
        updateValue(e: Event) {
            this.$emit('update:modelValue', (e.target as HTMLDivElement).innerText)
        }
    }
})
</script>

<style lang="scss" scoped>
div[contenteditable] {
    border: none;
    outline: none;
    font-size: 16px;
}
</style>