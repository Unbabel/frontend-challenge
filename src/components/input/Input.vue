<script>
	// Store
	import { mapState } from "vuex";

	export default {
		props: {
			value: {
				type: String,
			},
			placeholder: {
				type: String,
				default: "Insert text here",
			},
			customStyles: {
				type: String,
				default: "",
			},
			onChangeCallback: {
				type: Function,
			},
		},
		methods: {
			fitToContent() {
				this.$el.style.height = "5px";
				this.$el.style.height = this.$el.scrollHeight + "px";
			},
			onChangeHandler(ev) {
				this.fitToContent();
				this.onChangeCallback(ev.target.value);
			},
			getClasses() {
				const isColorBlindnessActive =
					this.$store.getters.isColorBlindnessActive;
				return `${this.$props.customStyles} ${
					isColorBlindnessActive && "colorBlindness"
				}`;
			},
		},
		mounted() {
			this.fitToContent();
		},
	};
</script>

<template>
	<textarea
		:class="getClasses()"
		:value="value"
		@input="onChangeHandler"
		:placeholder="placeholder"
	>
	</textarea>
</template>

<style scoped>
	textarea {
		width: 100%;
		border: 1px solid transparent;
		outline: none;
		resize: none;
	}

	textarea:focus {
		border: 1px solid var(--palette-lighter-grey);
		border-radius: 3px;
		background-color: var(--palette-dirty-white);
	}

	@media (max-width: 768px) {
		textarea {
			font-size: 0.9rem !important;
		}
	}
</style>
