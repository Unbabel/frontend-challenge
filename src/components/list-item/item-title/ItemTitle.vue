<script setup>
	// Components
	import Input from "../../input/Input.vue";

	defineProps({
		text: {
			type: String,
		},
	});

	const emit = defineEmits(["onChangeCallback"]);

	const onChangeHandler = (value) => emit("onChangeCallback", value);
</script>

<template>
	<Input
		customStyles="title-styles"
		:value="text"
		:onChangeCallback="onChangeHandler"
		placeholder="Insert title..."
	/>
</template>

<style scoped>
	.title-styles {
		font-family: "Montserrat";
		font-weight: 600;
		color: var(--palette-light-grey);
		max-height: 1.25rem;
		font-size: 1rem;
	}
</style>
