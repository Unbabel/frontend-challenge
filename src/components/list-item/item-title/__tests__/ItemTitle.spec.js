import Vuex from "vuex";
import { describe, it, expect, beforeEach, vi } from "vitest";
import { mount, createLocalVue, shallowMount } from "@vue/test-utils";

// Components
import ItemTitle from "../ItemTitle.vue";
import Input from "../../../input/Input.vue";

const localVue = createLocalVue();
localVue.use(Vuex);

const store = new Vuex.Store({});

describe("ItemTitle", () => {
	let wrapper;

	beforeEach(() => {
		// Arrange
		wrapper = shallowMount(ItemTitle, {
			propsData: {
				text: "This is some title text",
			},
		});
	});

	it("matches snapshot", () => {
		// Assert
		expect(wrapper.html()).toMatchSnapshot();
	});

	it("renders properly", () => {
		// Act
		const input = wrapper.findComponent(Input);

		// Assert
		expect(input.exists()).toBe(true);
	});

	it("input has correct classes", () => {
		// Act
		const input = wrapper.findComponent(Input);

		// Assert
		expect(input.props("customStyles")).toBe("title-styles");
	});

	it("input has correct placeholder", () => {
		// Act
		const input = wrapper.findComponent(Input);

		// Assert
		expect(input.props("placeholder")).toBe("Insert title...");
	});

	it("input has correct value", () => {
		// Act
		const input = wrapper.findComponent(Input);

		// Assert
		expect(input.props("value")).toBe("This is some title text");
	});

	it("should call onChangeHandler", async () => {
		// Arrange
		const spy = vi.spyOn(Input.methods, "onChangeHandler");
		wrapper = mount(ItemTitle, { localVue, store });

		// Act
		const input = wrapper.findComponent(Input);
		await input.trigger("input", { value: "Teste" });

		// Assert
		expect(spy).toBeCalled();
	});
});
