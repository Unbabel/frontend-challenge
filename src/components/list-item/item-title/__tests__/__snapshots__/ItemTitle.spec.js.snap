// Vitest Snapshot v1

exports[`ItemTitle > matches snapshot 1`] = `"<anonymous-stub data-v-52cd4611=\\"\\" value=\\"This is some title text\\" placeholder=\\"Insert title...\\" customstyles=\\"title-styles\\" onchangecallback=\\"[Function]\\"></anonymous-stub>"`;
